app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,306 @@
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.graphics.drawscope.rotate
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.drawText
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.rememberTextMeasurer
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.cyrcle.R
+import kotlin.math.*
+import kotlinx.coroutines.*
+
+@Composable
+fun WheelView(
+    modifier: Modifier = Modifier,
+    onSegmentLanded: ((String) -> Unit)? = null
+): () -> Unit {
+  data class Segment(val name: String, val color: Color, val probability: Float)
+
+  val segments = remember {
+    listOf(
+        Segment("Nothing", Color.Gray, 50f),
+        Segment("Legendary", Color(0xFFFFD700), 0.1f),
+        Segment("Common", Color.Green, 32f),
+        Segment("Rare", Color.Blue, 16f),
+        Segment("Epic", Color(0xFF800080), 1.9f))
+  }
+
+  val probabilityRanges = remember {
+    var currentValue = 0
+    segments.map { segment ->
+      val range = (segment.probability * 10).toInt()
+      val start = currentValue
+      currentValue += range
+      start until currentValue
+    }
+  }
+
+  var rotation by remember { mutableStateOf(0f) }
+  var isSpinning by remember { mutableStateOf(false) }
+  var targetRotation by remember { mutableStateOf(0f) }
+  var currentRotation by remember { mutableStateOf(0f) }
+  var spinStartTime by remember { mutableStateOf(0L) }
+  var pauseAfterSpin by remember { mutableStateOf(false) }
+  var expectedSegment by remember { mutableStateOf(0) }
+  val textMeasurer = rememberTextMeasurer()
+
+  fun getSegmentAtPointer(rotation: Float): Int {
+    val segmentAngle = 360f / segments.size
+    val normalizedAngle = (270 - rotation) % 360
+    return (normalizedAngle / segmentAngle).toInt()
+  }
+
+  fun determineTargetSegment(): Int {
+    val randomValue = (0..999).random()
+    Log.d("FAVORITE", "Random value generated: $randomValue")
+    for (i in probabilityRanges.indices) {
+      if (randomValue in probabilityRanges[i]) {
+        Log.d("FAVORITE", "Selected segment $i (${segments[i].name})")
+        return i
+      }
+    }
+    return 0
+  }
+
+  LaunchedEffect(Unit) {
+    while (true) {
+      if (!isSpinning && !pauseAfterSpin) {
+        rotation += -0.2f
```

Comment:
Not that important but this could simply be `rotation -= 0.2f`

Reply:
Of course...